contacts.py

```diff
@@ -0,0 +1,29 @@
+# Hacker rank problem link: https://www.hackerrank.com/challenges/contacts/problem
+
+trie = dict()

```

Comment:
That's a good implementation of a trie.

It would be better to make it a first class citizen by using object-oriented design:

class trie:
  def __init__(self):
    self.trie = dict()

  def add(word):
    # TODO: Insert the logic from add_contact.

  def number_of_matches(word):
    '''Returns the number of words matching the prefix |word| in the trie.''''
    # TODO: Insert the logic from  find_partial.
trie = trie()
def add_contact(word):
  trie.add(word)